src/workloads/query/MatchWithLargeExpression.yml

```diff
@@ -0,0 +1,150 @@
+- TemplateName: MatchExpressionWithOrClausesTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "unused"}}
+    Type: CrudActor
+    Database: *Database
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "ActivePhase", Default: -1}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: {^Parameter: {Name: "Repeat", Default: 1}}
+          Collection: *Collection
+          Operations:
+          - OperationName: aggregate
+            OperationCommand:
+              Pipeline: [
+                {$match: {$or: {^Array: {of: 
+                  {$and: [
+                    {a: {$gt: {^Inc: {start: 10}}}},
+                    {b: {$gt: {^Inc: {start: 10}}}}
+                  ]},
+                  number: {^Parameter: {Name: "NumberOfClauses", Default: -1}}
+                }}}}
+              ]
+
+Actors:
+# Clear any pre-existing collection state.
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        Collection: *Collection
+        Threads: 1
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: *DocumentCount
+        Document:
+          a: [0, 1]
+          b: [2, 3]
+          c: 4
+          d: 5
+          object: {}
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+
+- ActorFromTemplate:
+    TemplateName: MatchExpressionWithOrClausesTemplate
+    TemplateParameters:
+      Name: RunMatchExpressionOrWith50Clauses
+      Repeat: 1000
+      ActivePhase: 3
+      NumberOfClauses: 50
+
+- ActorFromTemplate:
+    TemplateName: MatchExpressionWithOrClausesTemplate
+    TemplateParameters:
+      Name: RunMatchExpressionOrWith100Clauses
+      Repeat: 1000
+      ActivePhase: 4
+      NumberOfClauses: 100
+
+- ActorFromTemplate:
+    TemplateName: MatchExpressionWithOrClausesTemplate
+    TemplateParameters:
+      Name: RunMatchExpressionOrWith1000Clauses
+      Repeat: 100
+      ActivePhase: 5
+      NumberOfClauses: 1000
+
+- ActorFromTemplate:
+    TemplateName: MatchExpressionWithOrClausesTemplate
+    TemplateParameters:
+      Name: RunMatchExpressionOrWith10000Clauses
+      Repeat: 10
+      ActivePhase: 6
+      NumberOfClauses: 10000
+
+- ActorFromTemplate:
+    TemplateName: MatchExpressionWithOrClausesTemplate
+    TemplateParameters:
+      Name: RunMatchExpressionOrWith50000Clauses
+      Repeat: 5
```

Comment:
This is interesting- are the later queries (5000 clauses) that much slower than the earlier ones that we can only run them 5 times, even if we have the optimization? How long does each one take to run?

Reply:
More like I would have leaned towards running them all the same number of times... but then its good to ahve as many datapoints as we can (for stability) in the faster cases. I'd say either 5 or 10 is fine, you can just keep it as is.

It seems like we have a non-linear performance degradation for larger clause sizes: is this correct?

Reply:
> It seems like we have a non-linear performance degradation for larger clause sizes: is this correct?

I'd say it's somewhat linear with the optimization, and way worse without:

Median time per op (in ms):
```
With optimization
100 - 5.173
1k  - 58.11
10k - 919.9
50k - 13500

Without optimization
100 - 5.809
1k  - 128.4
10k - 8122
50k - 232900
```